src/ucp/proto/proto_am.inl

```diff
@@ -296,7 +296,8 @@ ucs_status_t ucp_do_am_zcopy_multi(uct_pending_req_t *self, uint8_t am_id_first,
         req->send.lane = ucp_send_request_get_am_bw_lane(req);
         ucp_send_request_add_reg_lane(req, req->send.lane);
     } else {
-        req->send.lane = ucp_ep_get_am_lane(ep);
+        req->send.lane         = ucp_ep_get_am_lane(ep);
+        req->send.pending_lane = req->send.lane;
```

Comment:
why needed? it's done by ucp_request_pending_add()

Reply:
To avoid returning `UCS_OK` for the first fragment. This way I can assume that UCS_OK will not be returned for the first fragment if there is no resources on AM lane

Reply:
why it's needed for active messages? how is it different than eager zcopy for example?

Reply:
because if no resources returned  we will have to copy user header again and subtract its length from the req->send.length.
so we have to distinguish no_resource and all other cases to restore req->send.length

Reply:
can we update send.length only after we are sure it was sent?

Reply:
i think we can, but it will be different from handling in single_zcopy case.
btw, why not initialize  pending_lane? We do not need to return UCS_OK if send was failed because of no_resources and req->send.pending_lane is null lane

Reply:
it's not consistent with other usages of pending lane so makes the flow harder to understand. let's keep setting pending_lane in single place, as it is today.

Reply:
i did not get the point, but ok